embassy-stm32/src/flash/mod.rs

```diff
@@ -0,0 +1,401 @@
+use crate::pac;
+use crate::peripherals::FLASH;
+use core::convert::TryInto;
+use core::marker::PhantomData;
+use core::ptr::write_volatile;
+use embassy::util::Unborrow;
+use embassy_hal_common::unborrow;
+
+use embedded_storage::nor_flash::{
+    ErrorType, NorFlash, NorFlashError, NorFlashErrorKind, ReadNorFlash,
+};
+
+const FLASH_BASE: usize = 0x8000000;
+
+#[cfg(flash_l4)]
+mod config {
+    use super::*;
+    pub(crate) const FLASH_SIZE: usize = 0x100000;
```

Comment:
Is this the max size for each family? Ideally the size should be the one for the current chip.